kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -338,67 +338,89 @@ def foundry_prove(
+                init_term = kcfg.get_unique_init().cterm.kast
+                init_term = KDefinition__expand_macros(foundry.kevm.definition, init_term)
+                init_cterm = CTerm(init_term)
+
+                _LOGGER.info(f'Expanding macros in target state for test: {test}')
+                target_term = kcfg.get_unique_target().cterm.kast
+                target_term = KDefinition__expand_macros(foundry.kevm.definition, target_term)
+                target_cterm = CTerm(target_term)
+                kcfg.replace_node(kcfg.get_unique_target().id, target_cterm)
+
+                _LOGGER.info(f'Starting KCFGExplore for test: {test}')
+                with KCFGExplore(
+                    foundry.kevm,
+                    bug_report=br,
+                    kore_rpc_command=kore_rpc_command,
+                    smt_timeout=smt_timeout,
+                    smt_retry_limit=smt_retry_limit,
+                ) as kcfg_explore:
+                    _LOGGER.info(f'Computing definedness constraint for test: {test}')
+                    init_cterm = kcfg_explore.cterm_assume_defined(init_cterm)
+                    kcfg.replace_node(kcfg.get_unique_init().id, init_cterm)
+
+                    if simplify_init:
+                        _LOGGER.info(f'Simplifying KCFG for test: {test}')
+                        kcfg = kcfg_explore.simplify(test, kcfg)
+                kcfgs[test] = kcfg
+                KCFGExplore.write_cfg(test, kcfgs_dir, kcfg)
+
+        return parallel_kcfg_explore(
+            foundry.kevm,
+            kcfgs,
+            save_directory=kcfgs_dir,
+            max_depth=max_depth,
+            max_iterations=max_iterations,
+            workers=workers,
+            break_every_step=break_every_step,
+            break_on_jumpi=break_on_jumpi,
+            break_on_calls=break_on_calls,
+            implication_every_block=implication_every_block,
+            is_terminal=KEVM.is_terminal,
+            extract_branches=KEVM.extract_branches,
+            bug_report=br,
+            kore_rpc_command=kore_rpc_command,
+            smt_timeout=smt_timeout,
+            smt_retry_limit=smt_retry_limit,
+        )
 
-    return parallel_kcfg_explore(
-        foundry.kevm,
-        kcfgs,
-        save_directory=kcfgs_dir,
-        max_depth=max_depth,
-        max_iterations=max_iterations,
-        workers=workers,
-        break_every_step=break_every_step,
-        break_on_jumpi=break_on_jumpi,
-        break_on_calls=break_on_calls,
-        implication_every_block=implication_every_block,
-        is_terminal=KEVM.is_terminal,
-        extract_branches=KEVM.extract_branches,
-        bug_report=br,
-        kore_rpc_command=kore_rpc_command,
-        smt_timeout=smt_timeout,
-        smt_retry_limit=smt_retry_limit,
-    )
+    _LOGGER.info(f'Running setup functions in parallel: {list(setup_methods.values())}')
+    run_cfg_group(list(setup_methods.values()))
```

Comment:
We should be checking taht running the setup functions didn't result in any proof failures (never should, but who knows?) Basically, just check that `run_cfg_group` returns all `True` with:

```suggestion
    _LOGGER.info(f'Running setup functions in parallel: {list(setup_methods.values())}')
    results = run_cfg_group(list(setup_methods.values()))
    failed = (setup_cfg for setup_cfg, passed in results.items() if not passed)
    if failed:
        raise ValueError(f'Running setUp method failed for {len(failed)} contracts: {failed}')
```